### PR TITLE
Drop Down Overlay

### PR DESCRIPTION
**Pull request by ndickerson**
## **Description**

Using the CDK overlay template everywhere instead of our home grown overlay. Removing the need for appendToBody now that scrollStrategy is available. 

#### **Verify that...**

- [x] Any related demos where added and `npm start` still works
- [x] New demos work in `Safari`, `Chrome` and `Firefox`
- [x] `npm run lint` passes
- [x] `npm test` passes and code coverage is increased
- [x] `npm run compile` still works

##### **Screenshots**
overlay

### CODE REVIEW

File: src/platform/elements/data-table/cell-headers/data-table-header-cell.component.ts [hunk 1]
```diff
@@ -32,9 +32,9 @@ import { Helpers } from '../../../utils/Helpers';
         <i class="bhi-{{ labelIcon }} label-icon" *ngIf="labelIcon" data-automation-id="novo-data-table-header-icon"></i>
         <label data-automation-id="novo-data-table-label">{{ label }}</label>
         <div>
-            <button *ngIf="config.sortable" tooltipPosition="right" [tooltip]="labels.sort" theme="icon" [icon]="icon" (click)="sort()" [class.active]="sortActive" data-automation-id="novo-data-table-sort"></button>
-            <novo-dropdown *ngIf="config.filterable" side="right" appendToBody="true" parentScrollSelector=".novo-data-table-container" containerClass="data-table-dropdown" data-automation-id="novo-data-table-filter">
-                <button type="button" theme="icon" icon="filter" [class.active]="filterActive" (click)="focusInput()" tooltipPosition="right" [tooltip]="labels.filters"></button>
+            <button *ngIf="config.sortable" tooltipPosition="bottom" [tooltip]="labels.sort" theme="icon" [icon]="icon" (click)="sort()" [class.active]="sortActive" data-automation-id="novo-data-table-sort"></button>
```
Comment (jgodi) [inline src/platform/elements/data-table/cell-headers/data-table-header-cell.component.ts]: Conflict here? should still be right position for tooltip

File: src/platform/elements/chips/Chips.spec.ts [hunk 2]
```diff
@@ -4,7 +4,6 @@ import { TestBed, async } from '@angular/core/testing';
 // App
 import { NovoChipElement, NovoChipsElement } from './Chips';
 import { NovoChipsModule } from './Chips.module';
-//import { NovoPickerContainer } from '../picker/extras/picker-container/PickerContainer';
```
Comment (jgodi) [inline src/platform/elements/chips/Chips.spec.ts]: Just delete

File: src/platform/elements/data-table/data-table.component.ts [hunk 3]
```diff
@@ -1,39 +1,15 @@
 import {
-  ChangeDetectionStrategy,
```
Comment (jgodi) [inline src/platform/elements/data-table/data-table.component.ts]: Can you clean this file up? There were changes in here but hard to see

File: src/platform/elements/chips/Chips.spec.ts (hunk 2 above)
Comment (ndickerson) [reply src/platform/elements/chips/Chips.spec.ts]: It was deleted

**Pull request merged**